app/services/assignment_service.py

```diff
@@ -174,33 +199,36 @@ async def get_latest_due_date(self, assignment: AssignmentModel) -> datetime | N
             .first()
         
         return assignment.due_date + (latest_time.extra_time if latest_time.extra_time is not None else timedelta(0))
-    
-    async def get_master_notebook_name(self, assignment: AssignmentModel) -> str:
-        return self._compute_master_notebook_name(assignment.name)
-    
-    @staticmethod
-    def _compute_master_notebook_name(self, assignment_name: str) -> str:
-        return f"{ assignment_name }-prof.ipynb"
 
     """ Compute the default gitignore for an assignment. """
     async def get_gitignore_content(self, assignment: AssignmentModel) -> str:
-        master_notebook_name = await self.get_master_notebook_name(assignment)
-        return f"""### Python ###
-# Byte-compiled / optimized / DLL files
+        protected_files = ["\n".join(file) for file in await self.get_protected_files(assignment)]
+
+        return f"""### Defaults ###
 __pycache__/
 *.py[cod]
 *$py.class
-
-### Misc ###
-*.DS_Store
-*grades.csv
-{ master_notebook_name }
-{ assignment.name }-dist
-.ssh
-.ipynb_checkpoints
 *venv
-prof-scripts/*
+.ipynb_checkpoints
+.OTTER_LOG
+
+### Protected ###
+{ protected_files }
 """
+    
+    """
+    NOTE: File paths are not necessarily real files and may instead be globs.
+    NOTE: File paths are relative to `assignment.directory_path`.
+    """
+    async def get_protected_files(self, assignment: AssignmentModel) -> str:
```

Comment:
This doesn't need to be async 😂

Reply:
Yeah but there is a world in which this function needs to be async down the line